flink-connector-dynamodb/src/main/java/org/apache/flink/streaming/connectors/dynamodb/sink/DynamoDbWriteRequestType.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.dynamodb.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * Represents a DynamoDb Write Request type. The following types are currently supported
+ *
+ * <ul>
+ *   <li>PUT - Put Request
+ *   <li>DELETE - Delete Request
+ * </ul>
+ */
+@PublicEvolving
+public enum DynamoDbWriteRequestType {
+
+    // Note: Enums have no stable hash code across different JVMs, use toByteValue() for
+    // this purpose.
+    PUT((byte) 0),
+    DELETE((byte) 1);
```

Comment:
Is there a reason to not use `ordinal()` and define your own here?

Reply:
I think the only concern was keeping order - if order changes for different reasons, the ordinal changes as well - if it is not a concern, we can use the ordinal.
I sawit in the implementation for [Row Kind](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/types/RowKind.java#L106) you linked and assumed it was the reason

Reply:
I had linked the RowKind as an option to completely replace the enum we were using. However I am happy to stick with what we have got since they do not line up 100%. I agree that the byte mechanism is immune to ordering errors, but it seems like we could mitigate this with a good unit test. I will leave this decision up to you, I do not mind either way.